lib/carcost.cc

```diff
@@ -2,23 +2,80 @@
 #include "utils.h"
 
 namespace dsl {
+using namespace std;
 
-CarCost::CarCost(double ac, double eps) : ac(ac), eps(eps) {
+CarCost::CarCost(const CarGrid& grid, double ac, double eps)
+:grid(grid), use_twistnorm_metric(false),wt(Vector3d(0.1,1,2)), ac(ac),eps(eps){
+  assert(eps > 0 && eps < 1);
   assert(ac >= 0);
+}
+
+CarCost::CarCost(const CarGrid& grid, const Vector3d& wt, double eps )
+:grid(grid), use_twistnorm_metric(true), wt(wt), ac(0.1),eps(eps){
   assert(eps > 0 && eps < 1);
+  if(!(eps > 0 && eps < 1)){
+    cout<<"wrong eps in carcost"<<endl;
+    eps = 1e-6;
+  }
+
+  assert( (wt(0)>0) && (wt(1)>0) && (wt(2)>0) );
+  if(!((wt(0)>0) && (wt(1)>0) && (wt(2)>0))){
+    cout<<"wrong wt in carcost"<<endl;
+    this->wt << 0.1, 1, 2;
+  }
 }
 
-double CarCost::Real(const SE2Cell& a, const SE2Cell& b) const {
-  double d1 = (a.c.tail< 2 >() - b.c.tail< 2 >()).norm(); // position distance
-  double d2 = fabs(fangle(a.c(0) - b.c(0))); // orientation distance
-  return d1 + ac * d2;
+double CarCost::Real(const SE2Cell& a, const SE2Cell& b) const{
```

Comment:
It is indeed done incrementally and the Real() is only called from inside Connectivity. It's basically what the connectivity used to do. It made sense to have both the costs together to make sure that the heuristics is admissible.